### ISSUE TITLE
Reshape layer

### ISSUE THREAD

**Issue by opfromthestart** [coreylowman/dfdx#649]
Have a layer that can "undo" a Flatten layer, for models like auto-encoders that need to convert vectors into images.
Would be something like
```rust
struct Reshape<S: Shape>;
impl Module<Tensor<S2>> for Reshape<S> {
    fn try_forward(&self, inp: Tensor<S2>) -> Result<Tensor<S>, Err> {
        inp.try_reshape()
    }
}
```

**Comment by coreylowman** [comment]
I like it, good idea!

**Comment by coreylowman** [comment]
Closed by #666
